### ISSUE TITLE
Add a --no-output option

### ISSUE THREAD

**Issue by jeremyfa** [openfl/lime#525]
In order to be able to implement error highlighting/linting properly in editors, it would be nice to have a way to compile haxe files of a lime/openfl project without any output, just in order to get the haxe compilation errors after saving files in the project, and not only after running the project.

In the Atom lime plugin (https://atom.io/packages/lime), I initially tried to use the hxml arguments provided by `lime display` command, and ran haxe compiler with it (after having ensured there is a `--no-output` argument). However it seems that this hxml, normally used for code completion, can be different than the one used to actually build the project: when running haxe build with these args we are getting unexpected errors that we don't have when running `lime build ...` on the same project (related issue: https://github.com/openfl/openfl/issues/734).

It would be nice to have a way to either:
- Be able to run something like `lime build --no-output` that will just print haxe errors if there are any
- Be able to get the exact hxml argument used to build the project (and not the one used for code completion from `lime display`), so that we can use it to run haxe compilation with `--no-output` additional argument
- Ensure the hxml args used for display and build remain the same for a single project with the same options, allowing us to use the `lime display` arguments consistently not only for code completion, but also for compilation with no output, and any other task involving the haxe compiler on the project. **(preferred solution)**. But afaik currently, `lime display` arguments are forcing the use of the new code?
- Another solution I am not thinking about?

I am also open to suggestions if there is a better way to get haxe errors without building the whole project, assets etc...

Any thoughts?

**Comment by ibilon** [comment]
So if I understand correctly what you need is some sort of command that just update the templates (including the hxml) and run haxe on it with `--no-output`?

**Comment by jeremyfa** [comment]
Yes, something like this. Basically I need the haxe compiler output (to get and display compilation errors on the ide when there are any) without actually building the project. Ideally it should not change any file on disk, but I don't know if this is possible with lime?

**Comment by ibilon** [comment]
Well, ignoring the first time where the hxml/ApplicationMain(...) need to be created, we could let the hxml as is,
but that'd ignore the modifications to the project file.
And those are files generated by the build system, not used by the user and which would be modified by running the build command anyway.

So I guess that'd be fine?

**Comment by jeremyfa** [comment]
That sounds good yes!

**Comment by jgranick** [comment]
I think that adding `--no-output` support makes sense, I think we may pass this through to the Haxe compile process already, but different targets would just need to skip some post-build operations in this mode
